### DISCUSSION TITLE
Creating a link with Material UI - Add example in doc?

### DISCUSSION THREAD

**Discussion by nick-keller** [TanStack/router#235] [Q&A]
Hello,
I am using MUI as my components library, and I am trying to make a link.

# My code
```tsx
import { FC } from 'react'
import { Link as MuiLink, LinkProps as MuiLinkProps } from '@mui/material'
import {
  Link as LocationLink,
  LinkProps as LocationLinkProps,
} from 'react-location'

export const Link: FC<MuiLinkProps & LocationLinkProps> = ({
  children,
  ...rest
}) => {
  return (
    <MuiLink component={LocationLink} {...rest}>
      {children}
    </MuiLink>
  )
}
```

And somewhere else in my code:
```tsx
      <Link
        sx={{ mt: 2 }} // MUI prop
        to="/my-page" // React Location prop
      >
        My link
      </Link>
```

# The issue
That work just fine, but I get an warning in the console:
```
Warning: Function components cannot be given refs. Attempts to access this ref will fail. Did you mean to use React.forwardRef()?

Check the render method of `MuiButtonBaseRoot`.
```

Has anyone encountered and solved the issue?

# Documentation suggestion
Once I have a flawless solution, maybe it could be a great addition to the examples section of the doc! (happy to contribute)

**Comment by frontendlane** [comment]
This continues to be a problem in `@tanstack/react-router`.

`<Link>` from `@tanstack/react-router` cannot be given a `ref`, which is a [requirement for using Mui's `component` prop](https://mui.com/material-ui/guides/composition/#caveat-with-refs).

@tannerlinsley if I'm getting this right `@tanstack/react-router`'s `<Link>` needs to be built using `React.forwardRef` (instead of a regular functional component) for this to work.

**Reply by tannerlinsley** [reply]
https://github.com/TanStack/router/blob/6f0f747efdb4ab963c244d20b4b99de70883c605/packages/react-router/src/index.tsx#L265

**Comment by StevenACoffman** [comment]
See https://tanstack.com/router/latest/docs/framework/react/guide/custom-link#mui-example

### TanStack Router

TanStack Router supports custom links through the `createLink` helper function. More information about [`createLink`](https://tanstack.com/router/latest/docs/framework/react/guide/custom-link) can be found in their documentation.

```tsx
import { createLink, type LinkComponent } from '@tanstack/react-router'
import { Link as MUILink } from '@mui/material'

const CustomLink = createLink(MUILink)

function IndexPage() {
  return (
    <CustomLink to="/about" search={{ name: 'test' }}>
      Link
    </CustomLink>
  );
}
```
